File: cpp/prime/prime.cpp
#include <iostream>
#include <limits>
#include <cmath>

using namespace std;

int main()
{
    int num;
  
    while (true)
    {
        // prompting for input
        cout << "Enter an integer (-999 to stop): ";
        cin >> num;

        // input-check
        if (!cin)
        {
            cout << "Invalid input!" << endl;
            cin.clear(); 
            cin.ignore(numeric_limits <streamsize> :: max(), '\n');
            continue;
        }
        
        if (num == -999)
        {
            return 0;
        }
        else if (num <= 1 )
        {
            cout << "The number is not prime." << endl;
            continue;
        }
         
        if (num == 2 || num == 3) // in case of 2 and 3
        {
            cout << "The number is prime." << endl;
            continue;
        } 

        if (num % 2 == 0) // checks for even numbers 
        {
            cout << "The number is not prime." << endl;
            continue;
        }

        bool isPrime = true; // flag 

        int n = sqrt(num);

        for (int i = 3; i <= n; i +=2) // checks for non-prime
        {
            if (num % i == 0)
            {
                isPrime = false;
                break;
            }
        }

        if (isPrime)
        {
            cout << "The number is prime." << endl;
        }   
        else
        {
            cout << "The number is not prime." << endl;
        } 
    }
}

File: python/prime/prime.py
import sys
from math import sqrt

while True:
    # prompting for input
    num = input("Enter an integer (-999 to stop): ")

    # error-handling
    try:
        num = int(num)
    except ValueError:
        print("ERROR: Enter a valid integer!")
        continue

    # if input is valid

    if num == -999: # to stop
        sys.exit(0)

    isPrime = True # flag

    if num <= 1: 
        isPrime = False
    elif num != 2 and num % 2 == 0:
        isPrime = False    
    else:
        x = int(sqrt(num))
        # checking for divisors
        for i in range(3, x + 1, 2):
            if num % i == 0: # if a divisor is found
                isPrime = False
                break
        
    if isPrime:
        print("The number is prime.")
    else:
        print("The number is not prime.")
